dotnet/src/dotnetframework/GxClasses/Core/GXUtilsCommon.cs

```diff
@@ -3373,16 +3373,8 @@ public static string getTempFileName(string baseDir, string name, string extensi
 
 			lock (syncObj)
 			{
-
 				name = FixFileName(name, string.Empty);
-				if (string.IsNullOrEmpty(name))
-				{
-					fileName = tempFileName(baseDir, name, extension);
-				}
-				else
-				{
-					fileName = PathUtil.SafeCombine(baseDir, name + "." + extension);
-				}
+				fileName = tempFileName(baseDir, name, extension);
 				GxFile file = new GxFile(baseDir, fileName, fileType);
```

Comment:
I think that now we don't need to check if file exists anymore. 
We can safely assume it would not exist as GUID are unique. 

In Java implementation we don't have this additional check. 
In external storage it would mean an additional http request that will always return false.

Reply:
Done. You are right.
Lock(syncObj) is also unnecessary once the File.Exists is removed.